fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProgressIndicator.kt

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.ui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * UI displaying a circular progress indicator continuously loading.
+ *
+ * @param size The size of the progress indicator.
+ */
+@Composable
+fun IndeterminateProgressIndicator(
+    size: Dp = 24.dp,
+) {
+    CircularProgressIndicator(
+        modifier = Modifier.size(size),
```

Comment:
Should we hoist the modifier so the caller can position the indicator based on the layout?